blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.fetchTweet = this.fetchTweet.bind( this );
+			this.state = {
+				url: this.props.attributes.url,
+				html: '',
+				error: false,
+				fetching: false,
+			};
+		}
+		doFetch( url, setAttributes, setState ) {
+			setState( { fetching: true, error: false } );
+			jQuery.ajax( {
+				type: 'GET',
+				dataType: 'jsonp',
+				data: {},
+				timeout: 5000,
+				url: 'https://publish.twitter.com/oembed?url=' + encodeURI( url ),
+				error: function() {
+					setState( { fetching: false, error: true } );
```

Comment:
A big issue with asynchronous behaviors in components is that you can't always be sure that the component is still mounted by the time the network request completes (e.g. if a user removes the block). If that were to occur and this line is called, an error will occur because you're trying to set state on a component that doesn't exist anymore.

A component instance has an `isMounted` function, but this [has been discouraged as an anti-pattern](https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html).

In this case, what you will likely want to do is assign the return value of `jQuery.ajax` into an instance variable and abort the request if the component is unmounted:

```js
componentWillUnmount() {
	if ( this.fetchXHR ) {
		this.fetchXHR.abort();
		delete this.fetchXHR;
	}
}

doFetch() {
	this.fetchXHR = jQuery.ajax( {
		// ...
	} );
}